dicom_standard/table_utils.py

```diff
@@ -2,16 +2,52 @@
 Functions for low-level manipulation of standard tables,
 represented by a list-of-lists.
 '''
-from typing import List, Tuple, Dict
+from typing import List, Tuple, Dict, Callable, Optional, Match, Any
 from copy import copy
-from bs4 import Tag
+from bs4 import BeautifulSoup, Tag
 
+from dicom_standard import parse_lib as pl
 from dicom_standard import parse_relations as pr
 
 TableListType = List[List[Tag]]
+AttributeDictType = Dict[str, str]
 
 
-def table_to_dict(table: TableListType, row_names: List[str]) -> List[Dict[str, List[Tag]]]:
+def get_chapter_tables(standard: BeautifulSoup, chapter_id: str, validity_func: Callable[[Tag], Match]) -> Tuple[TableListType, List[Tag]]:
+    chapter_table_divs = pl.all_tdivs_in_chapter(standard, chapter_id)
+    filtered_table_divs = list(filter(validity_func, chapter_table_divs))
+    table_lists = list(map(tdiv_to_table_list, filtered_table_divs))
+    return (table_lists, filtered_table_divs)
+
+
+def tables_to_json(
+    tables: List[TableListType],
+    tdivs: List[Tag],
+    table_to_dict_func: Callable[[TableListType], List[Dict[str, List[Tag]]]],
+    get_table_with_metadata: Callable[[Tuple[TableListType, Tag]], Dict[str, Any]]
+) -> List[Dict[str, Any]]:
```

Comment:
You're doing a great job with adding type annotations as you go. For longer types like this, it could make sense to make a type variable, eg

```py
TableToDictFunc = Callable[[TableListType], List[Dict[str, List[Tag]]]]
```

Just for readability's sake